sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClient.java

```diff
@@ -1008,8 +1015,21 @@ private boolean isManagementToken(String lockToken) {
 
     private Mono<Void> updateDisposition(ServiceBusReceivedMessage message, DispositionStatus dispositionStatus,
         String deadLetterReason, String deadLetterErrorDescription, Map<String, Object> propertiesToModify,
-        ServiceBusTransactionContext transactionContext) {
+        ServiceBusTransactionContext transactionContext, ServiceBusErrorSource errorSource) {
```

Comment:
Why do we need the error source parameter? It can be inferred from the Disposition status.